spring-kafka-docs/src/main/asciidoc/retrytopic.adoc

```diff
@@ -151,6 +156,42 @@ public KafkaTemplate<String, Object> kafkaTemplate() {
 IMPORTANT: Multiple `@KafkaListener` annotations can be used for the same topic with or without manual partition assignment along with non-blocking retries, but only one configuration will be used for a given topic.
 It's best to use a single `RetryTopicConfiguration` bean for configuration of such topics; if multiple `@RetryableTopic` annotations are being used for the same topic, all of them should have the same values, otherwise one of them will be applied to all of that topic's listeners and the other annotations' values will be ignored.
 
+[[retry-topic-global-settings]]
+===== Configuring Global Settings and Features
+
+Since 2.9, the previous bean-based configuration approach has been deprecated.
+Instead, the `RetryTopicConfigurationSupport` class should be extended, the proper methods overridden, and imported in a `@Configuration` annotated class.
+An example follows:
+
+====
+[source, java]
+----
+
+@EnableKafka
+@Import(MyRetryTopicConfiguration.class)
+@Configuration
+public class KafkaConfiguration {
+}
+
+public class MyRetryTopicConfiguration extends RetryTopicConfigurationSupport {
+
+    @Override
+    protected void configureBlockingRetries(BlockingRetriesConfigurer blockingRetries) {
+        blockingRetries
+                .retryOn(MyBlockingRetriesException.class, MyOtherBlockingRetriesException.class)
+                .backOff(new FixedBackOff(3000, 3));
+    }
+
+    @Override
+    protected void manageNonBlockingFatalExceptions(List<Class<? extends Throwable>> nonBlockingFatalExceptions) {
+        nonBlockingFatalExceptions.add(MyNonBlockingException.class);
+    }
+}
+----
+====
+
+NOTE: When using this approach, the `@EnableRetryTopic` annotation is not necessary.
```

Comment:
It might not be relevant to this PR, but we need to investigate "and what will happen if we use both?".
Separate issue or so.

Reply:
In my integration tests they work well together - TBH I don't know in details how this works internally and why the duplicated imports don't trigger an exception for already having the beans. Maybe it's something from the test runtime?

Reply:
As I said: let's revise it in the separate issue!
Yes, it might be the fact that Spring allows to override beans by default.
Spring Boot does not starting with some version.
I have something like this in one test to verify that override cause the problem:
```
public class NoBeansOverrideAnnotationConfigContextLoader extends AnnotationConfigContextLoader {

	@Override
	protected void customizeBeanFactory(DefaultListableBeanFactory beanFactory) {
		beanFactory.setAllowBeanDefinitionOverriding(false);
	}

}
...
@ContextConfiguration(loader = NoBeansOverrideAnnotationConfigContextLoader.class)
```

Or like this to make it explicit in the specific test configuration:
```
@Bean
public static BeanFactoryPostProcessor beanFactoryPostProcessor() {
	return beanFactory -> ((DefaultListableBeanFactory) beanFactory).setAllowBeanDefinitionOverriding(false);
}
```

Reply:
Sure, makes sense. Thanks.

Reply:
I've changed the text to advise not to use both at the same time - that looks like the safest bet. As you mentioned, we can review this in a separate issue. Thanks.